File: src/main/Java/boj_step4/boj10810.java
import java.io.*;
import java.util.StringTokenizer;

public class boj10810 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer tokenizer = new StringTokenizer(reader.readLine());
        int n = Integer.parseInt(tokenizer.nextToken());
        int m = Integer.parseInt(tokenizer.nextToken());
        int[] basket = new int[n];
        while (m-- > 0) {
            tokenizer = new StringTokenizer(reader.readLine());
            int i = Integer.parseInt(tokenizer.nextToken()) - 1;
            int j = Integer.parseInt(tokenizer.nextToken()) - 1;
            int k = Integer.parseInt(tokenizer.nextToken());
            for (int index = i; index <= j; index++) basket[index] = k;
        }
        for (int ball: basket) writer.append(ball + " ");
        writer.close();
    }
}

File: src/main/kotlin/boj10810.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    var (n, m) = r.readLine()!!.split(" ").map { it.toInt() }
    var basket = IntArray(n)
    var ball = 0
    while (m-- > 0) {
        val (i, j, k) = r.readLine()!!.split(" ").map { it.toInt() }
        for(cnt in i..j) basket[cnt - 1] = k
    }
    for (i in 0 until basket.size) {
        w.write("${basket[i]} ")
    }
    w.close()
}